packages/keystone/runtimeStats.js

```diff
@@ -0,0 +1,171 @@
+const express = require('express')
+const camelCase = require('lodash/camelCase')
+
+const conf = require('@open-condo/config')
+const { getLogger } = require('@open-condo/keystone/logging')
+const metrics = require('@open-condo/keystone/metrics')
+
+const { DEFAULT_HEALTHCHECK_URL } = require('./healthCheck')
+
+const X_TARGET_OPTIONS_VAR_NAME = 'X_TARGET_OPTIONS'
+const ACCESS_TOKEN_VAR_NAME = 'RUNTIME_STATS_ACCESS_TOKEN'
+const ENABLED_VAR_NAME = 'RUNTIME_STATS_ENABLE'
+
+const IS_BUILD_PHASE = conf.PHASE === 'build'
+const IS_WORKER_PROCESS = conf.PHASE === 'worker'
+
+const logger = getLogger('runtime-stats')
+
+class RuntimeStatsMiddleware {
+    constructor ({
+        statsUrl = '/api/runtime-stats',
+    } = {}) {
+        this.statsUrl = statsUrl
+    }
+
+    async prepareMiddleware ({ keystone }) {
+        if (conf[ENABLED_VAR_NAME] !== 'true') {
+            logger.info({ msg: 'runtime stats disabled' })
+            return
+        }
+
+        if (!conf[X_TARGET_OPTIONS_VAR_NAME]) {
+            logger.warn({
+                msg: 'There are no options for x-target header. All queries will relate to the `other` group.',
+                data: { howToFix: `Add the ${X_TARGET_OPTIONS_VAR_NAME} variable to env. For example: 'condo-app,billing,cc-app'` },
+            })
+        }
+
+        const runtimeStats = {
+            activeRequestsIds: new Set(),
+            activeRequestsCountByType: {},
+            activeRequestsCountByTarget: {},
+            totalRequestsCount: 0,
+            totalRequestsCountByType: {},
+            totalRequestsCountByTarget: {},
+        }
+
+        if (!IS_BUILD_PHASE && !IS_WORKER_PROCESS) {
+            setInterval(function sendRuntimeMetrics () {
+                const { activeRequestsIds, ...otherStats } = runtimeStats
+
+                logger.info({
+                    msg: 'current values',
+                    runtimeStats: { activeRequestsIds: Array.from(activeRequestsIds.keys()), ...otherStats },
+                })
+
+                metrics.gauge({ name: 'runtimeStats.requestsCount.total', value: runtimeStats.totalRequestsCount })
+
+                for (const type of Object.keys(runtimeStats.totalRequestsCountByType)) {
+                    metrics.gauge({
+                        name: `runtimeStats.requestsCount.type.${camelCase(type)}.total`,
```

Comment:
is it lower camel case? like 'someWord'? and not 'SomeWord', just to verify

Reply:
I think we should use label instead?

Reply:
Added tags to a single metric